### DISCUSSION TITLE
CSR vs CSC

### DISCUSSION THREAD

**Discussion by Padarn** [pyg-team/pytorch_geometric#4248] [General]
Hey @rusty1s sorry need your help to clarify this line:

https://github.com/pyg-team/pytorch_geometric/blob/e4e43a9254cf136a57aa5ae1350675d59c6d72cb/torch_geometric/loader/utils.py#L43

To CSC seems to use CSR? 

From what I can tell (but I haven't fully understood it yet) this is a convention confusion on my end: CSR and CSC are swapped depending on if we want to represent the adjacency matrix as (row, col) = link from row to col or the reverse.

**Comment by rusty1s** [comment]
It is indeed confusing. In message passing, we generally need to make use of the transposed adjacency matrix to guarantee a message passing flow from source to destination nodes via `A @ X`. As such, we store the transposed adjacency matrix in `adj_t`. 

Note that the CSR representation of a transposed matrix is the CSC representation of the original matrix. As such, when sampling, we make use of the CSC representation of the original adjacency matrix.

**Reply by Padarn** [reply]
Okay understood, that makes sense. In this case, do you think in the sampler it would be less confusing if we stored the CSR components of `adj_t`? (I know practically this is just variable naming, but it might save from going back and forth between thinking about what representation is being used. 

Then again perhaps its not less confusing, but just how I somehow got myself thinking about it.

**Reply by rusty1s** [reply]
I think referring to that representation as CSC is better as this is indeed the CSC representation of `edge_index`.

**Reply by Padarn** [reply]
You mean that the `edge_index` specified in pytorch_geometric data is the COO format of the adjacency matrix, and so it matches to have the CSC format of the same matrix?

**Reply by rusty1s** [reply]
Yes :)

**Reply by Padarn** [reply]
Thanks, its clear now!
